### PR TITLE
 Text overflow in statusMessage panel in adminHomePage #4112 

### LINKED ISSUE #4112: Text overflow in statusMessage panel in adminHomePage
text overflow

As shown above, there is text overflow in statusMessage panel showing the join link in adminHomePage, when trying to add a new instructor.

### PR DESCRIPTION
**Pull request by mukesh14149**
Fixes #4112

### CODE REVIEW

**Comment by junhaoyap** [comment]
@mukesh14149  please correct your pull request name - look at the others and see the difference :)

**Comment by damithc** [comment]
@mukesh14149 pls post a screenshot.

**Comment by mukesh14149** [comment]
Earlier look


After fixing

**Comment by bamboo3250** [comment]
Isn't it hard to copy the join link if we hide the overflowed texts like that? Let's make it show full text.

**Comment by bamboo3250** [comment]
And please undo those files that are renamed.

**Comment by mukesh14149** [comment]
After
screenshot from 2016-03-03 20 13 43

**Comment by damithc** [comment]
I wonder how this change affects tables in other places.

**Comment by mukesh14149** [comment]
I think I am just fixing the size of the table, So, it shouldn't effect anywhere. Or Should I make those changes with id instead of class ?

**Comment by bamboo3250** [comment]
To be safe, let's just make changes in adminHomePage as the issue title mentioned only. I'm afraid that other pages may not look nice because of this.

screenshot 2016-03-04 11 17 28

**Comment by mukesh14149** [comment]
done, I use the id of that table.

**Comment by bamboo3250** [comment]
Awesome :+1: 
@damithc do you have any cosmetic request?

**Comment by damithc** [comment]
No further change requests from me.

**Comment by bamboo3250** [comment]
@mukesh14149 Could you try using `word-break : break-all;` for `td` inside the table then allocate 50% of the width for `Message` column. I think it may look nicer.

**Comment by bamboo3250** [comment]
And remove `table-layout: fixed;`. It distributes the width of 6 columns equally so it looks a little bit off to me.

**Comment by mukesh14149** [comment]
When allocating 50% width for message column, it does not look nice, i tried allocating width of 3 columns to message then it looks perfectly fine, so shall i change width to 3 columns or keep it at width of 6 columns.
6 columns width
final

3 columns width
final2

**Comment by bamboo3250** [comment]
Nice! 3-column width looks better, I think. The point is to minimize the height of the row while keeping everything look nice. You can see there is a big wasted space below the first 5 columns. If we have 10 rows like this, the page would look very long.
I think we still can make the `Message` column wider a bit by having `Institution` column smaller because institutions' names usually contain small words.

**Comment by mukesh14149** [comment]
@bamboo3250 done, i set it to 5-column width

**Comment by bamboo3250** [comment]
@mukesh14149 Could you upload a screenshot of it? :grin:

**Comment by mukesh14149** [comment]
finalstep

**Comment by mukesh14149** [comment]
@bamboo3250 instead of showing this big link why we don't  put a word and link it to this join link.So when a person click on this word that will open in another tab.

**Comment by mukesh14149** [comment]
@bamboo3250 instead of showing this big link why we don't put a word and link it to this join link. It will solve the issue of height of the row.

**Comment by damithc** [comment]
@mukesh14149 In your 3-column screenshot, content of the institution column is cut off? I don't want any content to get cut off. Try using longer values and confirm that no column content is cut off.

**Comment by mukesh14149** [comment]
@damithc i test it for "Indraprastha institute of information t" as you can see in the text field of institution above, it doesn't cut off.
For more simplicity i have a screenshot below.
its a 3-column width
damit

**Comment by bamboo3250** [comment]
@damithc Are you fine with putting the join link inside <a> tag instead? It indeed will solve this problem but also will make you have one more step, i.e. right-click on the link > copy link address.

**Comment by damithc** [comment]
Yes, we can try that.

**Comment by bamboo3250** [comment]
@mukesh14149 Okay, let's try your approach, putting the join link in a word then we can undo all changes in css and jsp files. You can find where we generate the message in `AdminInstructorAccountAddAction.java`.

**Comment by mukesh14149** [comment]
Yes i already did this, I just want to ask what should i write name of link, is that ok with "JoinLink" or do i write something else.
joinlink

**Comment by damithc** [comment]
`Join link`?

**Comment by mukesh14149** [comment]
I attach a link with word "link"
join

**Comment by bamboo3250** [comment]
I think now we can remove `word-break: break-all;` as well and the ':' should be '.'.

**Comment by mukesh14149** [comment]
done!
finallllll

File: src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java [hunk 1]
```diff
@@ -116,7 +116,7 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         
         List<InstructorAttributes> instructorList = backDoor.getInstructorsForCourse(courseId);
         String joinLink = logic.sendJoinLinkToNewInstructor(instructorList.get(0), data.instructorShortName, data.instructorInstitution);
-        data.statusForAjax = "Instructor " + data.instructorName + " has been successfully created with join link:<br>" + joinLink;
+        data.statusForAjax = "Instructor " + data.instructorName + " has been successfully created with join " + "<a href="+joinLink+">link.</a>";
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java]: While we are at it, can be shortened further:
Instructor John Doe created ([join link]())

**Comment by mukesh14149** [comment]
@damithc 
yothatisfinal

File: src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java [hunk 2]
```diff
@@ -116,7 +116,7 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         
         List<InstructorAttributes> instructorList = backDoor.getInstructorsForCourse(courseId);
         String joinLink = logic.sendJoinLinkToNewInstructor(instructorList.get(0), data.instructorShortName, data.instructorInstitution);
-        data.statusForAjax = "Instructor " + data.instructorName + " has been successfully created with join link:<br>" + joinLink;
+        data.statusForAjax = "Instructor " + data.instructorName + " created (" + "<a href="+joinLink+">join link</a>).";
```
Comment (bamboo3250) [inline src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java]: Could you add a space between operands and the operator `+`?

**Comment by mukesh14149** [comment]
Like that @bamboo3250 
Instructor John Doe created ( [join link](url) ).

**Comment by mukesh14149** [comment]
done!

**Comment by bamboo3250** [comment]
Sorry, we have an UI test for Admin Home Page. The change in the message makes it failed.
@mukesh14149 Could you make change to the test then run all tests to make sure everything pass? If you encounter timeout error, you can temporarily inscrease the timeout in `test.properties` file.

**Comment by mukesh14149** [comment]
ok i am doing.

**Comment by mukesh14149** [comment]
@bamboo3250 done!

**Comment by bamboo3250** [comment]
@mukesh14149 Good job :+1: Have you run all tests?

**Comment by mukesh14149** [comment]
Yes i run all tests.
Previously these test case failing but after changing in test file it goes fine.
screenshot from 2016-03-07 19 36 59

**Comment by mukesh14149** [comment]
Yes @bamboo3250 I run all tests, Earlier i was testing in firefox version 44.0.2 and it failing 50 test case but after downgrade it works fine.

**Comment by bamboo3250** [comment]
Please check for the AdminHomePageUiTest as well. I think it's failing due to the new message.

**Comment by mukesh14149** [comment]
Apologize for mistake, Some test cases are skipped, I get back soon after running all test again.

**Comment by mukesh14149** [comment]
Still AdminHomePageUiTest showing me this error, Below is expected code vs differ code image, I am not getting why it showing me this expected code because i only change message part.
screenshot from 2016-03-08 16 06 35
And  how much time should i select for timeout because in some case i get timeout error.

**Comment by wkurniawan07** [comment]
@mukesh14149 If I understand the error source correctly, syncing with the current `master` branch will fix it. 25-30 seconds is typically enough for dev server test.

**Comment by mukesh14149** [comment]
Thanks @bamboo3250 but still one case fail limited access i think.
screenshot from 2016-03-09 22 40 13

**Comment by wkurniawan07** [comment]
@mukesh14149 I'm not a Linux user so I'm not entirely sure, but it looks like a problem that has been addressed in our [troubleshooting guide](https://docs.google.com/document/d/1_p7WOGryOStPfTGA_ZifE1kVlskb1zfd3HZwc4lE4QQ/pub?embedded=true).

**Comment by mukesh14149** [comment]
@wkurniawan07 thanks for your help, i tried this but still getting same error. @bamboo3250  could you please run test, I think its my system problem.

**Comment by mukesh14149** [comment]
@bamboo3250 could you please review.

File: src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java [hunk 3]
```diff
@@ -110,7 +110,7 @@ private void testCreateInstructorAction() throws Exception {
                                         .withRegistrationKey(encryptedKey)
                                         .withInstructorInstitution(institute)
                                         .toAbsoluteString();
-        assertEquals("Instructor AHPUiT Instrúctör has been successfully created with join link:\n" + expectedjoinUrl,
```
Comment (bamboo3250) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java]: The `expectedjoinUrl` is not used anymore. Could you delete it? and followed by the `encryptedKey` as well.

**Comment by mukesh14149** [comment]
done.

File: src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java [hunk 4]
```diff
@@ -116,7 +116,7 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         
         List<InstructorAttributes> instructorList = backDoor.getInstructorsForCourse(courseId);
         String joinLink = logic.sendJoinLinkToNewInstructor(instructorList.get(0), data.instructorShortName, data.instructorInstitution);
-        data.statusForAjax = "Instructor " + data.instructorName + " has been successfully created with join link:<br>" + joinLink;
+        data.statusForAjax = "Instructor " + data.instructorName + " created (" + "<a href=" + joinLink + ">join link</a>)";
```
Comment (bamboo3250) [inline src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java]: Could you check this line again? I think it should be ` "<a href=\"" + joinLink + "\">join link</a>)"` (notice the escaped double quotes for the joinLink

Comment (bamboo3250) [inline src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java]: So that it will be `<a href="xxxxxx">join link</a>` in the html.

**Comment by mukesh14149** [comment]
done.

File: src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java [hunk 5]
```diff
@@ -103,14 +103,8 @@ private void testCreateInstructorAction() throws Exception {
...
         assertEquals(String.format(Const.StatusMessages.INSTRUCTOR_DETAILS_LENGTH_INVALID, Const.LENGTH_FOR_NAME_EMAIL_INSTITUTION), 
                      homePage.getMessageFromResultTable(1)); 
-        
-        String encryptedKey = StringHelper.encrypt(BackDoor.getKeyForInstructor(demoCourseId, instructor.email));
-        // use AppUrl from Config because the join link takes its base URL from build.properties
-        String expectedjoinUrl = Config.getAppUrl(Const.ActionURIs.INSTRUCTOR_COURSE_JOIN)
-                                        .withRegistrationKey(encryptedKey)
-                                        .withInstructorInstitution(institute)
-                                        .toAbsoluteString();
-        assertEquals("Instructor AHPUiT Instrúctör has been successfully created with join link:\n" + expectedjoinUrl,
```
Comment (damithc) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java]: @bamboo3250 we are weakening the test here. Is there no way to check the correctness of the registration key in the link?

Comment (bamboo3250) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java]: I guess we can use `verifyContains()` to check for the existence of the row in page source.
@wkurniawan07 do you have any suggestion?

Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java]: `element.getAttribute("innerHTML")` will do the trick.

Comment (bamboo3250) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java]: @wkurniawan07 Good idea. Thanks :+1: 

@mukesh14149 Let's do that. Now we bring `expectedjoinUrl` back then select the cell and check for its `innerHtml`.

**Comment by thyageshm** [comment]
@mukesh14149 please try to finish this one by this release (tomorrow)

**Comment by thyageshm** [comment]
closing due to inactivity

**Pull request closed without merging**